### DISCUSSION TITLE
Resources not loading after login

### DISCUSSION THREAD

**Discussion by riddhibatas-searce** [Sunbird-Ed/Community#189] [Installation] [answered]
Hello community,

We ran all the jenkins job successfully and then ran the post-install script as well. Every step worked for us, after post-installation we were able to login to the website.
But, after login, few requests are not getting fulfilled.

Below are the attached screenshots of the issue faced.
Any help would be appreciated :pray::pray:
Open to share logs and other screenshots if required for debugging the issue.

<img width="980" alt="Screenshot 2022-09-22 at 7 42 08 PM" src="https://user-images.githubusercontent.com/98751951/191770553-142ca9be-756e-4f7c-845d-d809758f49c6.png">

<img width="1440" alt="Screenshot 2022-09-22 at 6 44 30 PM" src="https://user-images.githubusercontent.com/98751951/191770482-04a8af1d-5173-493b-b8aa-47c15672d426.png">

**Comment by beepdot** [comment, marked as answer]
Please can you post details on which release you are setting up? Which document did you refer to setup Sunbird?

**Reply by riddhibatas-searce** [reply]
@keshavprasadms, 
I followed **release-4.6.0** release
Sunbird setup doc: http://docs.sunbird.org/latest/developer-docs/server-installation/build-and-deploy/

**Reply by beepdot** [reply]
I believe you might be having many API failures. From the screen shot, I can see you are getting channel as `undefined`.
Have you run the PostInstall jenkins job?

**Reply by beepdot** [reply]
Is the custodian org APIs successful?
https://github.com/project-sunbird/sunbird-devops/blob/master/ansible/roles/post-install/defaults/main.yml#L1632-L1635
https://github.com/project-sunbird/sunbird-devops/blob/master/ansible/roles/post-install/tasks/user_org.yaml#L2-L13

**Reply by riddhibatas-searce** [reply]
@keshavprasadms 
The postInstall jenkins job was successfully executed. We used the release tag **release-4.6.0_RC4** as mentioned in the documentation.

Also, the custodian org APIs are successful, not sure why we are getting undefined instead of org_id.
Should we rebuild any of the jobs?

<img width="710" alt="Screenshot 2022-09-23 at 4 55 32 PM" src="https://user-images.githubusercontent.com/98751951/191951691-c40b97ed-4fa6-4d4c-9cc6-3d2ecd5a2ee2.png">

**Reply by riddhibatas-searce** [reply]
Also @keshavprasadms, all the requests are served with **guest user**, it's just when we login from another user the requests are not served

**Reply by beepdot** [reply]
It looks like the `rootorgid` column value is probably `null` for the user who is logging in. Please verify this in cassandra db. Below is a sample query
```
EXPAND ON
SELECT id,rootorgid from sunbird.user where id = 'ENTER_THE_USER_ID_HERE';
```
Can you confirm if my guess is right?

**Reply by riddhibatas-searce** [reply]
@keshavprasadms yes you are correct, it worked after making the changes.
Thank you

**Reply by riddhibatas-searce** [reply]
@keshavprasadms we made the changes and were able to see the requests getting served. 

But when we go to Workspace page, we still get the **something went wrong, please try again later** for any button we click.
Can you please us with issue as well? Are there any jobs we missed on or any api which is causing the error?

<img width="1317" alt="Screenshot 2022-09-28 at 4 21 04 PM" src="https://user-images.githubusercontent.com/98751951/192761268-2f06518c-22a6-4115-9375-3aa86c0020ef.png">

**Reply by beepdot** [reply]
I will need some additional information like what API is failing, what is the response message and code as this is a very generic message which pops up for any type of issues. You should be able to get these from the browser Inspect -> Network tab.

**Reply by riddhibatas-searce** [reply]
@keshavprasadms Please have a look at the screenshots below,

Also, the content service is up and running on the cluster

<img width="847" alt="Screenshot 2022-09-30 at 11 28 19 AM" src="https://user-images.githubusercontent.com/98751951/193200571-615772aa-19c8-41b7-908f-c5a7ab60e449.png">


<img width="847" alt="Screenshot 2022-09-30 at 11 24 18 AM" src="https://user-images.githubusercontent.com/98751951/193200013-13f6a25e-bc55-40d8-bfc1-ffc98ebee8b9.png">

**Reply by beepdot** [reply]
Is the framework NCF created?
For Service Unavailable, see - https://github.com/orgs/Sunbird-Ed/discussions/115#discussioncomment-3677806

**Reply by riddhibatas-searce** [reply]
@keshavprasadms we have NCF12 framework currently running, can you please suggest how we can rectify this?

**Reply by beepdot** [reply]
Can you create the NCF framework?

**Reply by beepdot** [reply]
Also I would like to know how was NCF12 framework created? It looks like you have modified certain API's and created these yourselves? Or did you use the PostInstall job? I am unable to understand what all has been done / customised.

**Reply by riddhibatas-searce** [reply]
@keshavprasadms everything got created using the postinstall job. It's just that when ever we used to re-run the post-install job we just added a number after the variables so as to keep a track of which job got successful. Other than that there isn't anything that we changed

**Reply by riddhibatas-searce** [reply]
we also used to delete the previously created NCF from cassandra and elastic search

**Reply by beepdot** [reply]
You will have to either create the NCF framework or make your new framework as the default one for your channel. The API response `/api/channel/v1/read/:ID` shows the default framework.

**Reply by riddhibatas-searce** [reply]
Hi @keshavprasadms, it would be really helpful if you can guide me how to make the current framework as the default one for the channel.

**Reply by beepdot** [reply]
Before that I need to know the response of the above API in your environment. Please post the json response for the channel read API.

**Reply by riddhibatas-searce** [reply]
**For NCF12 we are getting** 

`{"id":"api.channel.read","ver":"1.0","ts":"2022-10-06T08:31:47.698Z","params":{"resmsgid":"514c5d20-4551-11ed-b342-2d1f9a2ccd67","msgid":"512b1980-4551-11ed-8f50-c58dcc56026d","status":"successful","err":null,"errmsg":null},"responseCode":"OK","result":{"channel":{"identifier":"NCF12","lastStatusChangedOn":"2022-09-22T11:51:35.316+0000","code":"NCF12","channel":"0136302433148928002","description":"NCF12 Framework","assetAdditionalCategories":[],"type":"K-12","languageCode":[],"createdOn":"2022-09-22T11:51:35.316+0000","objectType":"Framework","versionKey":"1663847495316","collectionPrimaryCategories":["Content Playlist","Course","Digital Textbook","Question paper"],"systemDefault":"No","contentPrimaryCategories":["Course Assessment","eTextbook","Explanation Content","Learning Resource","Practice Question 
Set","Teacher Resource","Exam Question"],"appId":"dev.dev.portal","name":"NCF12","primaryCategories":[],"additionalCategories":[],"lastUpdatedOn":"2022-09-22T11:51:35.316+0000","categories":[{"name":"Medium","relation":"hasSequenceMember","identifier":"ncf12_medium","description":null,"objectType":"CategoryInstance","status":"Live"},{"name":"Subject","relation":"hasSequenceMember","identifier":"ncf12_subject","description":null,"objectType":"CategoryInstance","status":"Live"},{"name":"GradeLevel","relation":"hasSequenceMember","identifier":"ncf12_gradelevel","description":null,"objectType":"CategoryInstance","status":"Live"},{"name":"Board","relation":"hasSequenceMember","identifier":"ncf12_board","description":null,"objectType":"CategoryInstance","status":"Live"}],"collectionAdditionalCategories":["Textbook","Lesson Plan"],"assetPrimaryCategories":["Asset","CertAsset","Certificate Template"],"contentAdditionalCategories":["Classroom Teaching Video","Concept Map","Curiosity Question Set","Experiential Resource","Explanation Video","Focus Spot","Learning Outcome Definition","Lesson Plan","Marking Scheme Rubric","Pedagogy Flow","Previous Board Exam Papers","TV Lesson","Textbook"],"status":"Live"}}}`



**For NCF, we are getting**

`{"id":"api.channel.read","ver":"1.0","ts":"2022-10-06T08:32:15.602Z","params":{"resmsgid":"61ee2d20-4551-11ed-b342-2d1f9a2ccd67","msgid":null,"status":"failed","err":"NOT_FOUND","errmsg":"Error! Node(s) doesn't Exists. | [Invalid Node Id.]: NCF"},"responseCode":"RESOURCE_NOT_FOUND","result":{"messages":null}}`

**Reply by beepdot** [reply]
This is the response of your framework. Please share me the response of your channel read API. Please look for this API url
`/api/channel/v1/read/:ID`. The `:ID` will be your root org id / custodian org id.

**Reply by riddhibatas-searce** [reply]
@keshavprasadms 

`{"id":"api.channel.read","ver":"1.0","ts":"2022-10-06T10:23:38.004Z","params":{"resmsgid":"f0f3cd40-4560-11ed-b342-2d1f9a2ccd67","msgid":"f0e882a0-4560-11ed-8f50-c58dcc56026d","status":"successful","err":null,"errmsg":null},"responseCode":"OK","result":{"channel":{"identifier":"0136302433148928002","lastStatusChangedOn":"2022-09-22T11:51:16.060+0000","code":"0136302433148928002","frameworks":[{"name":"NCF12","relation":"hasSequenceMember","identifier":"NCF12","description":"NCF12 Framework","objectType":"Framework","status":"Live","type":"K-12"}],"consumerId":"7909c24e-49f6-41b1-90c6-9e48b13a38b0","channel":"in.ekstep","description":"Default Channel","assetAdditionalCategories":[],"languageCode":[],"createdOn":"2022-09-22T11:51:16.060+0000","objectType":"Channel","versionKey":"1663847498324","collectionPrimaryCategories":["Content Playlist","Course","Digital Textbook","Question paper"],"contentPrimaryCategories":["Course Assessment","eTextbook","Explanation Content","Learning Resource","Practice Question Set","Teacher Resource","Exam Question"],"appId":"dev.dev.portal","name":"sunbird","primaryCategories":[],"additionalCategories":[],"lastUpdatedOn":"2022-09-22T11:51:38.324+0000","collectionAdditionalCategories":["Textbook","Lesson Plan"],"assetPrimaryCategories":["Asset","CertAsset","Certificate Template"],"contentAdditionalCategories":["Classroom Teaching Video","Concept Map","Curiosity Question Set","Experiential Resource","Explanation Video","Focus Spot","Learning Outcome Definition","Lesson Plan","Marking Scheme Rubric","Pedagogy Flow","Previous Board Exam Papers","TV Lesson","Textbook"],"status":"Live","defaultFramework":"NCF"}}}`

**Reply by beepdot** [reply]
In the above output I can see this
`"defaultFramework":"NCF"`
Please update the channel's default value to your framework. Try this API - https://github.com/project-sunbird/sunbird-devops/blob/master/ansible/roles/post-install/tasks/knowledge_platform_tasks.yaml#L110-L119 and check if it updates the framework value.

**Reply by beepdot** [reply]
https://github.com/project-sunbird/sunbird-devops/blob/master/ansible/roles/post-install/tasks/knowledge_platform_tasks.yaml#L121-L139

**Reply by riddhibatas-searce** [reply]
hey @keshavprasadms, associated the framework with channel and it worked. Thank you!
